### PR TITLE
Track custom event for Azure Search requests

### PR DESCRIPTION
**Pull request by WolfyUK**
### Context
[AB#239253](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/239253) [AB#229670](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/229670)

### Change proposed in this pull request
Includes test coverage for `SearchConnection`

### Guidance to review 
Using the search suggester to locate a school, trust or LA should log the 'suggest' search criteria. Creating a custom comparator set for a school or trust should log the 'search' search criteria. These should both be visible in the Transaction (not timeline) view in App Insights as Custom Events linked to the parent Azure Search dependency and source API request, e.g.:

image

image

### Checklist (add/remove as appropriate)
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running ~~locally~~ in `d17` feature environment
- [ ] ~~You have reviewed with UX/Design~~

### CODE REVIEW

File: platform/src/apis/Platform.Api.Establishment/Configuration/Services.cs [hunk 1]
```diff
@@ -49,16 +52,27 @@ internal static void Configure(IServiceCollection serviceCollection)
...
-            .AddSingleton<ISearchConnection<ComparatorSchool>>(new SearchConnection<ComparatorSchool>(searchEndpoint, searchCredential, ResourceNames.Search.Indexes.SchoolComparators))
-            .AddSingleton<ISearchConnection<ComparatorTrust>>(new SearchConnection<ComparatorTrust>(searchEndpoint, searchCredential, ResourceNames.Search.Indexes.TrustComparators))
+            .AddSingleton<ISearchConnection<LocalAuthority>>(x => new SearchConnection<LocalAuthority>(
+                new SearchClient(searchEndpoint, ResourceNames.Search.Indexes.LocalAuthority, searchCredential),
+                x.GetService<ITelemetryService>()))
+            .AddSingleton<ISearchConnection<School>>(x => new SearchConnection<School>(
+                new SearchClient(searchEndpoint, ResourceNames.Search.Indexes.School, searchCredential),
+                x.GetService<ITelemetryService>()))
+            .AddSingleton<ISearchConnection<Trust>>(x => new SearchConnection<Trust>(
+                new SearchClient(searchEndpoint, ResourceNames.Search.Indexes.Trust, searchCredential),
```
Comment (jrabbott) [inline platform/src/apis/Platform.Api.Establishment/Configuration/Services.cs:62]: [Suggestion] could possibly add the SearchClient as a Keyed Singleton?


https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addkeyedsingleton

Comment (jrabbott) [reply platform/src/apis/Platform.Api.Establishment/Configuration/Services.cs:62]: Ignore me - it's not going to be that helpful

Comment (WolfyUK) [reply platform/src/apis/Platform.Api.Establishment/Configuration/Services.cs:62]: Looks funky! I'll have a read

Comment (WolfyUK) [reply platform/src/apis/Platform.Api.Establishment/Configuration/Services.cs:62]: &lt;sigh&gt; https://github.com/microsoft/ApplicationInsights-dotnet/issues/2879

Comment (WolfyUK) [reply platform/src/apis/Platform.Api.Establishment/Configuration/Services.cs:62]: I've done this anyway, as it makes it a little easier to read even taking into consideration the issue above. Just finishing validation.

**Review by jrabbott — APPROVED** [review]

**Pull request merged**